### PR TITLE
Introduce clearer boundry between extensions

### LINKED ISSUE #7471: Split internal typings of Bundled and non bundled legacy extensions
Motivation:
- The requirements for creation are different
- The source/discovery mechanisms are different

### PR DESCRIPTION
**Pull request by Nokel81**
- Bundled extensions are always enabled, and are always compatible
- Have bundled extensions be loaded asyncronously to support typescript dynamic import (which is typed) as opposed to require

### CODE REVIEW

File: packages/core/src/common/protocol-handler/router.ts [hunk 1]
```diff
@@ -209,7 +209,7 @@ export abstract class LensProtocolRouter {
       return name;
     }
 
-    if (!this.dependencies.extensionsStore.isEnabled(extension)) {
+    if (!extension.isBundled && !this.dependencies.extensionsStore.isEnabled(extension.id)) {
```
Comment (jim-docker) [inline packages/core/src/common/protocol-handler/router.ts]: If it is bundled and not enabled that would be an internal error? I guess you prevent this elsewhere.

**Comment by jim-docker** [comment]
Still reviewing but noticed this while testing:


https://user-images.githubusercontent.com/40840436/219488889-2457f7f1-d80f-46dc-befd-030d3c9ffc16.mov

**Comment by Nokel81** [comment]
Thanks! Will look into that

**Comment by jim-docker** [comment]
Do we really need to introduce this this close to release?

**Comment by Nokel81** [comment]
Maybe not. Might indeed be better to hold off until after. This was motivated by the need for documentation so it can hold off.

**Comment by Nokel81** [comment]
> Still reviewing but noticed this while testing:

Seems like it is merely a state data issue.

File: packages/core/src/renderer/components/+extensions/extensions.tsx [hunk 2]
```diff
@@ -4,114 +4,66 @@
...
-import extensionInstallationStateStoreInjectable from "../../../extensions/extension-installation-state-store/extension-installation-state-store.injectable";
-import type { ExtensionInstallationStateStore } from "../../../extensions/extension-installation-state-store/extension-installation-state-store";
 import Gutter from "../gutter/gutter";
 
+const ExtensionsNotice = () => (
+  <Notice className={styles.notice}>
+    <p>
+      {"Add new features via Lens Extensions. Check out the "}
+      <a
+        href={`${docsUrl}/extensions/`}
```
Comment (jim-docker) [inline packages/core/src/renderer/components/+extensions/extensions.tsx]: I noticed this is 404 now, maybe
```
     href={`${docsUrl}/extensions/lens-extensions`}
```
?

File: packages/core/src/extensions/extension-loader/extension-loader.ts [hunk 3]
```diff
@@ -203,17 +171,19 @@ export class ExtensionLoader {
     const extension = this.extensions.get(lensExtensionId);
 
     assert(extension, `Must register extension ${lensExtensionId} with before enabling it`);
```
Comment (jim-docker) [inline packages/core/src/extensions/extension-loader/extension-loader.ts]: I know this is old but there seems to be a word missing or extra?

**Review by jim-docker — COMMENTED** [review]
just need to test...

File: packages/core/src/common/protocol-handler/router.ts (hunk 1 above)
Comment (Nokel81) [reply packages/core/src/common/protocol-handler/router.ts]: If it is `bundled` then it is always enabled and cannot be disabled.

**Review by gabriel-mirantis — APPROVED** [review]

**Pull request merged**